src/makeCalculator.js

```diff
@@ -4,7 +4,68 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    add(value) {
+      this.result += value;
+
+      return this;
+    },
+
+    subtract(value) {
+      this.result -= value;
+
+      return this;
+    },
+
+    multiply(value) {
+      this.result *= value;
+
+      return this;
+    },
+
+    divide(value) {
+      if (value !== 0) {
+        this.result /= value;
+      } else {
+        throw new Error("Can't divide by 0!");
+      }
+
+      return this;
+    },
+
+    operate(operation, value) {
+      if (value === undefined) {
+        throw new Error('No value given');
+      }
+
+      switch (operation) {
+        case this.add:
+          this.add(value);
+          break;
+        case this.subtract:
+          this.subtract(value);
+          break;
+        case this.multiply:
+          this.multiply(value);
+          break;
+        case this.divide:
+          this.divide(value);
+          break;
+        default:
+          throw new Error('Unknown operation');
+      }
```

Comment:
The switch statement is unnecessary here. You can just call the operation function directly with the value. This will allow the calculator to work with any function, not just its own methods. This is a code quality issue.